kernelci/storage/__init__.py

```diff
@@ -33,13 +34,23 @@ def credentials(self):
         """Credentials data"""
         return self._credentials
 
+    def _connect(self):
```

Comment:
This should also be declared as an abstract method.
```suggestion
   @abc.abstractmethod
    def _connect(self):
```

Reply:
No because it doesn't need to be implemented.

Reply:
Yes, I saw that it is not implemented in `StorageBackend`.